File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if(argc == 2)
    {
        int k = atoi(argv[1]);
        string p = get_string("plaintext: ");
        printf("ciphertext: ");

        for(int i = 0, n = strlen(p); i < n; i++)
        {
            if(isalpha(p[i]))
            {
                if(isupper(p[i]))
                {
                    int a = (int)p[i] - 65;
                    a += k;
                    a = a % 26;
                    a += 65;
                    printf("%c", a);

                } else if (islower(p[i]))
                {
                    int a = (int)p[i] - 97;
                    a += k;
                    a = a % 26;
                    a += 97;
                    printf("%c", a);
                }

            } else
            {
                printf("%c", p[i]);
            }
        }
        printf("\n");

    } else
    {
        printf("You forgot to enter a key...\n");
        printf("Arguments Entered: %i , Arguments Expected: 1 \n", argc - 1);
    }
}

File: pset6/caesar.py
import sys
from cs50 import get_string

if len(sys.argv) == 2:
    k = int(sys.argv[1])
    p = get_string("plaintext: ");
    print("ciphertext: ", end="");

    for i in p:
        if i.isalpha:
            if i.isdigit():
                print(i, end="")
            if i.isupper():
                a = ord(i) - 65
                #print(f"       {a}")
                a += k
                a = a % 26
                a += 65
                print(chr(a), end="")
            elif i.islower():
                a = ord(i) - 97
                a += k
                a = a % 26
                a += 97
                print(chr(a), end="")
        else:
            print(i, end="")


else:
    print("You forgot to enter a key... Or you did something else wrong")


print()